File: src/main/java/com/tastyload/tastyload/placeGroup/entity/PlaceGroupEntity.java
package com.tastyload.tastyload.placeGroup.entity;

import com.tastyload.tastyload.users.UserEntity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "place_group")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PlaceGroupEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private UserEntity user;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "like_count")
    private int likeCount;

    @Column(name = "visibility")
    private boolean visibility;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
}

File: backend/src/place-groups/entities/place-group.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToOne,
  JoinColumn,
  OneToMany,
} from 'typeorm';
import { User } from '../../users/entities/user.entity';
import { PlaceGroupItem } from './place-group-item.entity';

@Entity('place_groups')
export class PlaceGroup {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ nullable: false })
  name: string;

  @Column({ nullable: true })
  description: string;

  @ManyToOne(() => User, { lazy: true })
  @JoinColumn({ name: 'user_id', referencedColumnName: 'uuid' })
  user: User;

  @CreateDateColumn({ name: 'created_at' })
  createdAt: Date;

  @UpdateDateColumn({ name: 'updated_at' })
  updatedAt: Date;

  @OneToMany(() => PlaceGroupItem, (item) => item.placeGroup)
  items: PlaceGroupItem[];
}